### PR TITLE
Small fixes fox proxy and remote

### PR DESCRIPTION
**Pull request by RABijl**

### CODE REVIEW

File: lib/CL/devices/proxy/pocl_proxy.cpp [hunk 1]
```diff
@@ -1402,8 +1402,11 @@ pocl_proxy_setup_metadata (cl_device_id device, cl_program program,
   proxy_device_data_t *d = (proxy_device_data_t *)device->data;
   cl_program proxy_prog = (cl_program)program->data[program_device_i];
 
-  if (!(d->backend->provides_metadata || d->backend->supports_il))
+  // return if there is no metadata and we are not building using a binary
+  if (!(d->backend->provides_metadata ||
```
Comment (pjaaskel) [inline lib/CL/devices/proxy/pocl_proxy.cpp:1406]: check format

**Review by pjaaskel — CHANGES_REQUESTED** [review]

File: lib/CL/devices/proxy/pocl_proxy.cpp [hunk 2]
```diff
@@ -1402,7 +1402,9 @@ pocl_proxy_setup_metadata (cl_device_id device, cl_program program,
   proxy_device_data_t *d = (proxy_device_data_t *)device->data;
   cl_program proxy_prog = (cl_program)program->data[program_device_i];
 
-  if (!(d->backend->provides_metadata || d->backend->supports_il))
+  // Return if there is no metadata and we are not building using a binary.
+  if (!(d->backend->provides_metadata ||
+        (d->backend->supports_il && program->program_il_size > 0)))
```
[the remark this answers is not in the document]
Comment (RABijl) [reply lib/CL/devices/proxy/pocl_proxy.cpp]: that is what the formatter determined was appropriate. I've moved it back a bit

[the remark this answers is not in the document]
Comment (jansol) [reply lib/CL/devices/proxy/pocl_proxy.cpp]: The formatter makes sense, it's aligns with the column after the innermost parenthesis around the `||` where the line was split.

[the remark this answers is not in the document]
Comment (pjaaskel) [reply lib/CL/devices/proxy/pocl_proxy.cpp]: OK, I'd have expected to align the if parenthesis.

**Review by pjaaskel — APPROVED** [review]

**Pull request merged**